pmd-core/src/main/java/net/sourceforge/pmd/cpd/XMLRenderer.java

```diff
@@ -119,7 +121,8 @@ private Element addFilesToDuplicationElement(Document doc, Element duplication,
             mark = iterator.next();
             final Element file = doc.createElement("file");
             file.setAttribute("line", String.valueOf(mark.getBeginLine()));
-            file.setAttribute("path", mark.getFilename());
+            String filenameXml10 = StringEscapeUtils.unescapeXml(StringEscapeUtils.escapeXml10(mark.getFilename()));
```

Comment:
StringEscapeUtils escape basic chars like "<" and remove invalid chars like U+000C. But the DOM impl already escapes "<" by itself in attribute values, but doesn't remove invalid chars. In order to avoid double escaping, this escapes it once with StringEscapeUtils and unescapes it again - result is, that only invalid chars are removed (no escaping).